### ISSUE TITLE
Buttons on the confirm dialog and the question dialog are unselectable on macOS 13

### ISSUE THREAD

**Issue by leaffeather** [beeware/toga#2104]
### Describe the bug

As the title shows, I cannot make any selection on a question dialog or a confirm dialog on macOS 13.

### Steps to reproduce

`python

if await self.main_window.confirm_dialog(strConfirmDialogTitle, strConfirmDialogMsg) == True: #question_dialog does not work as well

    do_something()

else:

    return

`

### Expected behavior

Fix it

### Environment

- Operating System: macOS 13
- Python version: 3.8.17
- Software versions: 
  - Briefcase:
  - Toga: 0.3.1
  - ...

### Logs

```

```

**Comment by freakboy3742** [comment]
You're going to need to provide more evidence to support your claim that this isn't working.

The Toga repository has an [example "dialogs" app](https://github.com/beeware/toga/tree/main/examples/dialogs) that works fine in my testing on macOS 13.3.1, and has done for many macOS releases before that. 

Can I ask you to please provide a full reproduction case that demonstrates the problem you're seeing. 

It would also be helpful if you clarify what "can't make a selection" means - do you mean that the dialog doesn't display? That it doesn't respond to button clicks? That it doesn't close when you *do* click on it?

**Comment by leaffeather** [comment]
<img width="321" alt="无标题" src="https://github.com/beeware/toga/assets/134275609/17601cb0-18e9-4df1-b8ad-b9ea93e06021">
You can refer to the code in my project:https://github.com/leaffeather/wordgets/blob/main/app.py at Line 2014. I run the macOS 13 in vmware. The code can work normally on Windows and Android.

**Comment by freakboy3742** [comment]
When providing a reproduction case, it's a *lot* more helpful to provide a *minimal* reproduction case. You know your code; I don't. And, I'm not especially enthusiastic about the idea of running a large codebase on my own machine when I don't know what the code is doing. 

Reducing a problem to a minimal reproduction case is an important part of the debugging process - and, in the process you'll often discover the source of the problem (or, at least, narrow it down significantly).

In this case, I suspect that narrowing down the problem to a minimal case will reveal that the problem is that `SyncDBToLocal` is being [invoked with an `asyncio.run()` call inside startup](https://github.com/leaffeather/wordgets/blob/main/app.py#L674C17-L674C54). This means you're creating a standalone asyncio event loop - one that will be competing with the event loop for your actual application. One run loop has created the dialog, the other run loop is processing the events that would allow the dialog to respond to the mouse.

This *might* work on some platforms, or in very specific circumstances, because of platform-specific variations on the exact startup process of an app, and/or the way asyncio integration with the GUI event loop is implemented. However, if it works, that's accidental. You should assume that once you've got an App instance, you have a run loop, and *that* is the run loop that you use for all async behaviour.

I suspect that if you replace your `asyncio.run()` invocations with `asyncio.ensure_future()`, you will get the behaviour you're looking for. `ensure_future()` will also invoke the co-routine on the event loop, but won't create a *new* event loop when doing so.

**Comment by leaffeather** [comment]
I tried asyncio.ensure_future() and there is an another issue raised: The question/confirm dialog runs only once if there is no network and I want to retry.

**Comment by freakboy3742** [comment]
That sounds like it's a logic error with your app, not a bug in Toga; on that basis, I'm going to close this ticket.

If you want assistance debugging your app, please open a [discussion topic](https://github.com/beeware/toga/discussions/new?category=debugging-help). Keep in mind that "please fix my code" isn't likely to get a good response; specific questions about *minimal* examples will be a lot more productive.
